web-transport-quinn/src/session.rs

```diff
@@ -558,8 +558,11 @@ impl web_transport_trait::Session for Session {
         Self::close(self, code, reason.as_bytes());
     }
 
-    async fn closed(&self) -> Self::Error {
-        Self::closed(self).await
+    async fn closed(&self) -> Result<(), Self::Error> {
+        match Self::closed(self).await {
+            SessionError::ConnectionError(quinn::ConnectionError::LocallyClosed) => Ok(()),
```

Comment:
You could maybe also return Ok for `ApplicationClosed(_)` (not sure).

Reply:
`ApplicationClosed` means the peer closed the connection via `session.close(code, reason)`. Maaaybe we could return Ok((code, reason)) instead of () but they definitely shouldn't be swallowed.